docs-ref-services/storage.md

```diff
@@ -16,19 +15,79 @@ The Azure Storage libraries for Java provide classes for working with data in yo
+    if (!ex.getErrorCode().equals(BlobErrorCode.CONTAINER_ALREADY_EXISTS)) {
+        throw ex;
+    }
+}
+
+/* Upload the file to the container */
+BlobClient blobClient = containerClient.getBlobClient("my-remote-file.jpg");
+blobClient.uploadFileFile("my-local-file.jpg");
+```
+
+For more examples, review the [Client Library README](https://github.com/Azure/azure-sdk-for-java/blob/azure-storage-blob_12.0.0/sdk/storage/azure-storage-blob/README.md).
+
+## Available packages
+
+ The following table describes the recommended versions of the storage client library for Java.
 
 | Library version | Supported services | Maven | Reference | Source, Readme, Examples |
 |----------------------|------------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|-----------------------------------------------------------------------------------------------------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
-| Version 12 (preview) | Blob, Queue, and File | [Blob](https://search.maven.org/artifact/com.azure/azure-storage-blob/)<br />[Queue](https://search.maven.org/artifact/com.azure/azure-storage-queue/)<br />[File](https://search.maven.org/artifact/com.azure/azure-storage-file/) | [Version 12 reference](https://docs.microsoft.com/java/api/overview/azure/storage?view=azure-java-preview) | [Blob](https://github.com/Azure/azure-sdk-for-java/tree/master/sdk/storage/azure-storage-blob)<br />[Queue](https://github.com/Azure/azure-sdk-for-java/tree/master/sdk/storage/azure-storage-queue)<br />[File](https://github.com/Azure/azure-sdk-for-java/tree/master/sdk/storage/azure-storage-file) |
-| Version 8 | Blob, Queue, File, and Table | [All services](https://mvnrepository.com/artifact/com.microsoft.azure/azure-storage) | [Version 8 reference](https://docs.microsoft.com/java/api/overview/azure/storage/client?view=azure-java-stable) | [All services](https://github.com/azure/azure-storage-java/tree/legacy-master) |
+| Version 12 | Blob and Queue | [Blob](https://search.maven.org/artifact/com.azure/azure-storage-blob/)<br />[Queue](https://search.maven.org/artifact/com.azure/azure-storage-queue/) | [Blob](https://azuresdkdocs.blob.core.windows.net/$web/java/azure-storage-blob/12.0.0/index.html)<br/>[Queue](https://azuresdkdocs.blob.core.windows.net/$web/java/azure-storage-queue/12.0.0/index.html) | [Blob](https://github.com/Azure/azure-sdk-for-java/tree/azure-storage-blob_12.0.0/sdk/storage/azure-storage-blob)<br />[Queue](https://github.com/Azure/azure-sdk-for-java/tree/azure-storage-blob_12.0.0/sdk/storage/azure-storage-queue) |
```

Comment:
I'd add links to Quickstarts on docs - they are updated

Reply:
Not sure what you mean there.  The heading is "Source, Readme, Examples" - and all those are in GitHub.

Reply:
Right, I did not think about the layout. maybe another column? I just think that would be a good pointer.